cpp/src/io/parquet/parquet_common.hpp

```diff
@@ -92,6 +92,7 @@ enum class Encoding : uint8_t {
   RLE_DICTIONARY          = 8,
   BYTE_STREAM_SPLIT       = 9,
   NUM_ENCODINGS           = 10,
+  UNDEFINED               = 255,
```

Comment:
UNDEFINED means no encoding is requested?

Reply:
Yes. I was hoping to use `thrust::optional` on the device, but the column descriptor lives in a struct in shared mem, which apparently is a no go. If I switch from string to enum on the metadata, I can just use that enum rather than repurposing the Encoding enum.  Will still need a "not set" value, but that's easier now since the new enum doesn't have to be `uint8_t`. (Unless, siomehow, the new `cuda::std::optional` works where `thrust::optional` did not).

Reply:
Do we then ditch the optional in the encode selection API and just default to UNDEFINED? No reason to have two "empty" states.

Reply:
But...but...`optional` 😭 I'll do a test to see if it can be optional on device (not very hopeful), and if it can't, then I'll switch. How do you feel about `NOT_SET` rather than `UNDEFINED`?